examples/stable_diffusion_xl/gm/models/trainer_factory.py

```diff
@@ -76,6 +79,10 @@ def construct(self, x, *tokens):
         noised_input = self.loss_fn.get_noise_input(x, noise, sigmas)
         w = append_dims(self.denoiser.w(sigmas), x.ndim)
 
+        if self.snr_gamma is not None:
+            snr_gamma = ops.ones_like(w) * self.snr_gamma
+            w = ops.stack((w, snr_gamma), axis=0).min(axis=0)
```

Comment:
I think it's better to extract the loss weighting modules from `TrainOneStepCell`. Otherwise, maintaining them will be challenging. Please refer to [Stability-AI repo](https://github.com/Stability-AI/generative-models/blob/9d759324e914de6c96dbd1468b3a4a50243c6528/sgm/modules/diffusionmodules/loss_weighting.py).

Reply:
The original weighting is the same as Stability-AI, see [here](https://github.com/mindspore-lab/mindone/blob/master/examples/stable_diffusion_xl/gm/modules/diffusionmodules/denoiser_weighting.py#L25)

Reply:
Yes, so why not adding SNR weighting there as well?